File: 374-Guess Number Higher or Lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        left, right = 1, n+1
 
        while True:
            guessed = left + (right-left)//2
            res = guess(guessed)
            if res == -1:
                right = guessed
            elif res == 1:
                left = guessed
            else:
                return guessed

File: JavaScript/374-Guess Number Higher or Lower.js
/** 
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	            -1 if num is lower than the guess number
 *			             1 if num is higher than the guess number
 *                       otherwise return 0
 * var guess = function(num) {}
 */

/**
 * @param {number} n
 * @return {number}
 */
var guessNumber = function(n) {
    let left = 1, right = n+1;
    while (true){
        const guessed = left + ((right-left)>>1);
        const res = guess(guessed);
        if (res === -1) right = guessed;
        else if (res === 1) left = guessed;
        else return guessed;
    }
};
